app/models/driver.rb

```diff
@@ -0,0 +1,51 @@
+class Driver < ApplicationRecord
+	has_many :trips, dependent: :destroy
+
+	validates :name, presence: true
+	validates :vin, presence: true
+
+	def average_rating
+		sum_rating = 0.0
+		rated_rides = 0
+
+		self.trips.each do |trip|
+			if trip.rating != nil
+				sum_rating += trip.rating
+				rated_rides += 1
+			end
+		end
+
+		self.trips.length > 0 ? (sum_rating / rated_rides).round(2) : sum_rating
```

Comment:
You can simplify this using a few `Enumerable` methods:

```suggestion
		rated_rides = trips.reject { |trip| trip.rating.nil? }
		sum_rating = rated_rides.sum { |trip| trip.rating }

		rated_rides.length > 0 ? (sum_rating / rated_rides.length).round(2) : 0
```